query-service-impl/src/main/java/org/hypertrace/core/query/service/function/DistinctCountFunctionTransformation.java

```diff
@@ -0,0 +1,79 @@
+package org.hypertrace.core.query.service.function;
+
+import static org.hypertrace.core.attribute.service.v1.AttributeKind.TYPE_BOOL_ARRAY;
+import static org.hypertrace.core.attribute.service.v1.AttributeKind.TYPE_DOUBLE_ARRAY;
+import static org.hypertrace.core.attribute.service.v1.AttributeKind.TYPE_INT64_ARRAY;
+import static org.hypertrace.core.attribute.service.v1.AttributeKind.TYPE_STRING_ARRAY;
+
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.core.Single;
+import java.util.List;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.core.attribute.service.cachingclient.CachingAttributeClient;
+import org.hypertrace.core.attribute.service.v1.AttributeKind;
+import org.hypertrace.core.query.service.AbstractQueryTransformation;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Function;
+import org.slf4j.Logger;
+
+@Slf4j
+public class DistinctCountFunctionTransformation extends AbstractQueryTransformation {
```

Comment:
How about we make this class more generic for transforming any multi valued attribute?

Reply:
I see, so something like `MULTIVALUE` transformation class which takes care of all these kind of transformations ?

Reply:
refactored